backend/services/news/src/datasources/News.ts

```diff
@@ -134,19 +154,48 @@ export default class News extends dbUtils.KnexDataSource {
       if (!article) throw new UserInputError('id did not exist');
 
       return {
-        article: convertArticle(article),
+        article: await this.convertArticle(article, ctx.user?.keycloak_id),
         uploadUrl: uploadUrl?.presignedUrl,
       };
     });
   }
 
-  removeArticle(ctx: context.UserContext, id: UUID): Promise<gql.Maybe<gql.RemoveArticlePayload>> {
+  removeArticle(ctx: context.UserContext, id: UUID): Promise<gql.Maybe<gql.ArticlePayload>> {
     return this.withAccess('news:article:delete', ctx, async () => {
       const article = await dbUtils.unique(this.knex<sql.Article>('articles').where({ id }));
       if (!article) throw new UserInputError('id did not exist');
       await this.knex<sql.Article>('articles').where({ id }).del();
       return {
-        article: convertArticle(article),
+        article: await this.convertArticle(article, ctx.user?.keycloak_id),
+      };
+    });
+  }
+
+  toggleLikeArticle(ctx: context.UserContext, id: UUID): Promise<gql.Maybe<gql.ArticlePayload>> {
```

Comment:
I'd recommend making this idempotent (calling the same endpoint multiple times has the same effect), since it is usually easier to reason around. Thereby splitting this in to likeArticle and unlikeArticle, the logic should become simpler too.

Reply:
If you also add a unique constraint to article + member (table.unique(['article_id', 'member_id'])) you won't need to check if an article is liked beforehand, just try it :)

Reply:
I think it already is unique so will try and remove the check